source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -195,6 +193,13 @@ void ProxyFilter::addHostAddressToFilterState(
                         StreamInfo::FilterState::LifeSpan::Request);
 }
 
+void ProxyFilter::onDnsResolutionFail() {
+  decoder_callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DnsResolutionFailed);
+  decoder_callbacks_->sendLocalReply(Http::Code::ServiceUnavailable,
```

Comment:
@Augustyniak @buildbreaker @goaway @jpsim @Reflejo FYI, now we will have a more explicit error for DNS failures from the forward proxy filter. Rather than the `upstream failure` error we have gotten used to.